feat(retry tx): Initialize retry-tx command PE-1246 (#275)
ardriveapp/ardrive-cli: README.md

}
```

### Retrying a Failed File Data Transaction (Public Unbundled Files Only)<a id="retry-tx"></a>

Arweave data upload transactions are split into two phases: transaction posting and chunks uploading. Once the transaction post phase has been completed, you've effectively "paid" the network for storage of the data chunks that you'll send in the next stage.

If your system encounters an error while posting the transaction, you can retry posting the transaction for as long as your tx_anchor is valid ([learn more about tx_anchors here][tx_anchors]). You may retry and/or resume posting chunks at any time after your transaction has posted. The ArDrive CLI allows you to take advantage of this Arweave protocol capability.

> Using the CLI, when the transaction post has succeeded but the chunk upload fails due to the process crashing the data transaction's ID could be lost. But a user has a few options to recover this ID. If it is their last transaction, they can quickly recover this transaction ID with the `ardrive last-tx -w /path/to/wallet` command. Another option us to use an Arweave gateway GQL endpoint to search for transactions that belong to their wallet. Or they could check out their latest transactions with a third party tool like [ViewBlock][viewblock].

Reviewer: ```suggestion
Using the CLI, when the transaction post has succeeded but the chunk upload step fails, the data transaction's ID could be lost. There are a few options to recover this ID. If the failed transaction is the most recent one sent from a wallet, the transaction ID can be quickly recovered with the `ardrive last-tx -w /path/to/wallet` command. Other options for finding the partially uploaded transaction's ID include: 
- Using an Arweave gateway GQL http endpoint to search for transactions that belong to the wallet. See this [Arweave GQL Guide][https://gql-guide.vercel.app/#owners] for more info.
- Browse the recent  transactions associated with the wallet via a block explorer tool like [ViewBlock][viewblock].
```

> In order to re-seed the chunks to an unbundled ArFS data transaction, a user must have the data transaction ID, the original file data, and a destination folder ID or a valid file ID for the file. With this information simply use the `retry-tx` command:

Reviewer: ```suggestion
In order to re-seed the chunks for an unbundled ArFS data transaction, a user must have the data transaction ID, the original file data, and either a destination folder ID or a valid file ID for the file. Supply that information to the `retry-tx` command like so:
```

```shell
ardrive retry-tx --transaction-id { Data Transaction ID } --parent-folder-id { Destination Folder ID }  --local-path /path/to/file  --wallet-file /path/to/wallet
```

> **Note: Retry feature is currently only available for PUBLIC unbundled file transactions. It is also perfectly safe to mistakenly re-seed the chunks of a healthy transaction, the transaction will remain stable and the wallet will not be re-charged.**

Reviewer: ```suggestion
**Note: Retry feature is currently only available for PUBLIC unbundled file transactions. It is also perfectly safe to mistakenly re-seed the chunks of a healthy transaction, the transaction will remain stable and the wallet balance will not be affected.**
```

With just that header information, a transaction accepted to the network, can be mined into a block, and will become optimistically accessible at various endpoints on the gateway including GQL. This all happens irregardless of whether the chunks of data arrive or not.

If the process of uploading chunks were to be interrupted (e.g: network failure, killing the process, system crash), the transaction will be left with incomplete chunks of data. And the transaction owner's wallet would be charged for the transaction based on the header information.

#### Information Required to Retry a Transaction

In order to re-seed the chunks to an unbundled ArFS data transaction, a user must have the data transaction ID, the original file data, and a destination folder ID or a valid file ID for the file.

When an upload fails, generally this transaction ID information would be lost. But a user can quickly recover this transaction ID with the `ardrive last-tx -w /path/to/wallet` command.

#### Retrying with ArDrive CLI

> With the data transaction ID and the file data is recovered a file simply use the `retry-tx` command:

Reviewer 2: I'm having a hard time to understand this phrase. I feel it doesn't make sense, maybe because English is not my main language.
